src/screens/SendToken/SendTokenAmount.js

```diff
@@ -175,23 +181,24 @@ class SendTokenAmount extends React.Component<Props, State> {
     });
   };
 
-  handleChange = (value: Object) => {
-    this.setState({ value });
+  handleChange = async (value: Object) => {
+    const gasLimit = await this.getGasLimit();
+    this.setState({ value, gasLimit });
   };
 
   handleFormSubmit = () => {
     this.formSubmitted = true;
     const txFeeInWei = this.getTxFeeInWei();
     const value = this._form.getValue();
     const { navigation } = this.props;
+    const { gasLimit } = this.state;
     const transactionSpeed = this.getTxSpeed();
-    const gasPrice = txFeeInWei.div(GAS_LIMIT).toNumber();
-
+    const gasPrice = txFeeInWei.div(gasLimit).toNumber();
```

Comment:
If the user is offline or there're some problems fetching the gas - the result will be `Infinity` as you're trying to divide by 0

Reply:
If user is offline or any problems occur then it it will return our default 500k gas limit.

Reply:
services/assets.js:

```
return provider.estimateGas({
    from,
    to,
    data,
    value,
  })
    ...
    .catch(() => DEFAULT_GAS_LIMIT);
```